README.md

```diff
@@ -1 +1,588 @@
+```
+
+If the user types the search keyword "javascript" in the input, instead of firing 10 API requests (one request for every character in the word "javascript"), we only fire one request; when the user has finished typing.
+
+However, and I’m sorry to disappoint, this is not how LitElement does things, but important to illustrate the point of "batching". Before we go deeper into that, we’re going to require some knowledge about...
+
+## THE EVENT LOOP!
+
+The event loop... can be a pretty elusive subject to understand. I'll try to keep it brief here with a basic example, but if you're interested in learning more about how the event loop works, here are some incredible resources that I cannot recommend enough:
+
+- [Tasks, Microtasks, Queues and Schedules](https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/) by [Jake Archibald](https://twitter.com/jaffathecake)
+- [In The Loop](https://www.youtube.com/watch?v=cCOL7MC4Pl0) by [Jake Archibald](https://twitter.com/jaffathecake)
+- [JavaScript Visualized](https://dev.to/lydiahallie/javascript-visualized-promises-async-await-5gke#tasks) by [Lydia Hallie](https://twitter.com/lydiahallie)
+
+Fortunately, where we're headed, we don't need to understand all the ins and outs of the event loop, we're really only interested in _microtasks_ for our case. Consider the following piece of code (... that often does the rounds on Twitter polls, and has confused many a developer before). What do you think is the order the `console.log` statements are called?
+
+```js
+console.log(1);
+
+Promise.resolve().then(() => {
+  console.log(2);
+});
+
+console.log(3);
+```
+
+Here's the answer, the order of the resulting console logs of this code is:
+
+```js
+// 1
+// 3
+// 2
+```
+
+We'll have [Jake Archibald](https://twitter.com/jaffathecake) explain to us why exactly this happens:
+
+> The microtask queue is processed after callbacks as long as no other JavaScript is mid-execution, and at the end of each task. Any additional microtasks queued during microtasks are added to the end of the queue and also processed.
+
+If we go back to our code snippet, we can see that the following happens:
+
+- `console.log(1);` is called, and logged to the console
+- We schedule a new _microtask_ with `Promise.resolve().then(() => {})`, but at this point in time, JavaScript has not finished executing; it's not time to actually process microtasks just yet
+- JavaScript continues executing code, because we're not done yet
+- `console.log(3);` is called, and logged to the console
+- JavaScript has finished executing! The browser can now process any microtasks we may have
+- `console.log(2);` is called, and logged to the console.
+
+So why is this important? How does this allow us to use microtasks to _batch_ work? Let's take a look at a practical example:
+
+```js
+/* We declare and initialize a variable to keep track of whether or not an update has already been requested */
```

Comment:
since you're going to later on anyways, why not adopt class syntax already here?